TrafficCapture/coreUtilities/src/main/java/org/opensearch/migrations/tracing/IInstrumentationAttributes.java

```diff
@@ -0,0 +1,109 @@
+package org.opensearch.migrations.tracing;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.metrics.DoubleHistogram;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.api.metrics.LongHistogram;
+import io.opentelemetry.api.metrics.LongUpDownCounter;
+import io.opentelemetry.api.trace.Span;
+import lombok.NonNull;
+
+import java.time.Duration;
+import java.util.ArrayList;
+
+public interface IInstrumentationAttributes {
+    AttributeKey<Boolean> HAD_EXCEPTION_KEY = AttributeKey.booleanKey("hadException");
+
+    IInstrumentationAttributes getEnclosingScope();
+
+    default Span getCurrentSpan() { return null; }
+
+    default AttributesBuilder fillAttributes(AttributesBuilder builder) {
+        return builder;
+    }
+
+    Throwable getObservedExceptionToIncludeInMetrics();
+
+    default @NonNull Attributes getPopulatedMetricAttributes(AttributesBuilder attributesBuilder) {
+        final var e = getObservedExceptionToIncludeInMetrics();
+        return e == null ? attributesBuilder.build() : attributesBuilder.put(HAD_EXCEPTION_KEY, true).build();
+    }
+
+    default Attributes getPopulatedSpanAttributes() {
+        return getPopulatedSpanAttributes(Attributes.builder());
+    }
+
+    default Attributes getPopulatedSpanAttributes(AttributesBuilder builder) {
+        return getPopulatedSpanAttributesBuilder(builder).build();
+    }
+
+    default AttributesBuilder getPopulatedSpanAttributesBuilder(AttributesBuilder builder) {
```

Comment:
It seems a bit strange here that the passed in AttributesBuilder's values will be overriden if set in any enclosingScope, initially I was thinking it would have priority over any enclosing values since presumably it's set by the caller in the current scope.

Reply:
Hmmmm - thanks.  That's a good point & not really what I should be intending to do.  This is a throwback from when different types of attributeBuilders were passed in here.

Reply:
I've deleted the AttributesBuilder as an argument.  Now it would be up to the caller to set the values on the AttributeBuilder AFTER filling them (from all of the enclosing scopes)